Changes in visitor.hpp to make compliant with user-defined data types

jbcolli2 opened the pull request:
This pull request makes changes to the file `visitors.hpp`.  These changes allow minieigen to be extended for use with user-defined data types.

There were two issues when extending minieigen for user-defined data types. 
**1.** On lines 121 and 122, there is a call to `std::abs`.  This has been changed to

```
using namespace std; 
return abs(num)<=absTol || num!=-0; 
```

This allows `std::abs` to be used when it is defined, but also allows user-defined `abs` to be used

**2.** In multiple places `boost::is_complex` is used to check if the data type is a complex type.  This is changed to `Eigen::NumTraits::IsComplex`.  This allows the check to work for user-defined data types which define the NumTraits appropriately.

jbcolli2 committed:
Remove std::abs from code
Use Eigen::NumTraits instead of boost::is_complex to check if data type is complex

Makes minieigen more compliant with user defined data types

ofloveandhate commented:
it looks like some temp files got added.  can you remove them?

e.g. build/temp.macosx-10.10-x86_64-2.7/src/double-conversion/bignum-dtoa.o

jbcolli2 committed:
Remove std::abs from code
Use Eigen::NumTraits instead of boost::is_complex to check if data type is complex

Makes minieigen more compliant with user defined data types

jbcolli2 commented:
Thanks Dan, that should clean it up.
